buildchain/buildchain/docker_command.py

```diff
@@ -0,0 +1,214 @@
+# coding: utf-8
+
+"""Provides Python commands for Docker Python client.
+
+Supply callable objects with prepared configuration for use
+with `doit`'s Python actions.
+"""
+
+import copy
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import docker                            # type: ignore
+from docker.types import Mount           # type: ignore
+import doit                              # type: ignore
+from doit.exceptions import TaskError    # type: ignore
+
+from buildchain import constants
+
+
+DOCKER_CLIENT : docker.DockerClient = docker.from_env()
+
+
+# The call method is not counter as a public method
+# pylint: disable=too-few-public-methods
+class BuildCommand:
+    """Docker build callable"""
+
+    def __init__(
+        self,
+        tag: str,
+        path: Path,
+        dockerfile: Path,
+        args: Dict[str, Any]
+    ):
+        self.tag = tag
+        self.path = str(path)
+        self.dockerfile = str(dockerfile)
+        self.args = args
+
+    def __call__(self) -> Union[TaskError, bool]:
+        try:
+            DOCKER_CLIENT.images.build(
+                tag=self.tag,
+                path=self.path,
+                dockerfile=self.dockerfile,
+                buildargs=self.args
+            )
+        except (docker.errors.BuildError, docker.errors.APIError) as err:
+            return TaskError(err)
+        return True
```

Comment:
Can be removed, as `None` is treated as success (return type would become `Optional[TaskError]`:

> For successful completion it must return one of:
> -    True
> -    None
> -    a dictionary
> -    a string

See http://pydoit.org/tasks.html#python-action

Ditto below.

Reply:
Pylint disagrees, since we need the explicit `TaskError` return, it demands no implicit returns.

Reply:
Fair enough, then `return None` (`return True` makes you expect `return False` in case of error, which is not the case here).